plugins/pencilblue/controllers/actions/salesforce/change_password.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const passport = require('koa-passport');
+const request = require('request-promise');
+
+module.exports = function SalesforceChangePasswordControllerModule(pb) {
+
+    //dependencies
+    /**
+     * Changes an user password
+     * @class SalesforceChangePasswordController
+     * @constructor
+     */
+    class SalesforceChangePasswordController extends pb.BaseController {
+        render(cb) {
+            this.sanitizeObject(this.body);
+            this.changePassword(cb);
+        }
+
+        async changePassword(cb) {
+            try {
+                const user = this.ctx.session.authentication.user;
+                if (user && user.salesforce) {
+                    const url = `${user.salesforce.authorize.instance_url}/services/data/v25.0/sobjects/User/${user.salesforce.profile.id}/password`;
```

Comment:
Minor: I think in a future refactor you could centralize several parts of this functionality (API call, error handling), due to the similarities between changing an email and changing a password

Reply:
Yes please, going back to the above comment about changing the folder structures and where this code lives.  We could have . a base SalesForceCredentailsController or something that abstracts a lot of this.  

I envision a BaseContorller who has two children - A BaseApiController and a BaseViewController.  Then all things stem from that.  The Base API controller could handle all the sending responses and such that we seem to repeat a lot.